yellowbrick/classifier/confusion_matrix.py

```diff
@@ -103,6 +108,20 @@ def score(self, X, y, sample_weight=None, percent=True):
         """
         y_pred = self.predict(X)
 
+
+        if self.label_mapping:
+            # if a mapping is passed to class apply it here.
+            y = character_mapping(y, self.label_mapping)
+            y_pred = character_mapping(y_pred, self.label_mapping)
+
+        if len(set(y_pred)) == len(set(self.classes_)):
+            if set(y_pred) != set(self.classes_):
+                self._label_encoder = LabelEncoder()
+                self._label_encoder.fit(self.classes_)
+
+                y = self._label_encoder.inverse_transform(y)
+                y_pred = self._label_encoder.inverse_transform(y_pred)
```

Comment:
I get what you're trying to do here, however I'm wondering if the logic might be a tiny bit off. I'd like to discuss what I think the inputs to both `y_pred` and `self.classes_` are to see what you think. 

`self.classes_` can be either:

1. a user supplied list of classes that is converted into a `np.array`
2. the underlying encoding by the estimator, stored in `self.estimator.classes_`, also a `np.array`. 

In both cases `classes_` is a collection of string values (names). In the `self.estimator.classes_` case, we assume that `classes_[0]` maps to an encoded class `0` and so on and so forth, however it also means that we're expecting `y_pred` to contain strings! If `self.estimator.classes_` are integers, it means that `y_train` was encoded and therefore `y_pred` should be encoded as well. 

And that brings us to `y_pred` which will either be:

1. A vector of ints if `y_train` was encoded 
2. A vector of strings if it wasn't 

So my worry here is that label encoding the `classes_` might actually produce a false mapping depending on what the user passes to the constructor. (Of course, I'm not sure, and I'm happy to discuss it). 

Additionally, this method should also help us take care of "slicing" -- that is when the user suggests a subset of classes to display in the confusion matrix. 

Any thoughts?